sw/host/opentitansession/src/main.rs

```diff
@@ -96,6 +113,119 @@ fn parse_command_line(opts: Opts, mut args: ArgsOs) -> Result<Opts> {
     Ok(opts)
 }
 
+#[derive(serde::Serialize, serde::Deserialize)]
+pub struct SessionStartResult {
+    port: u16,
+}
+
+/// Spawn a child process, passing all the same arguments to the child, letting it instantiate a
+/// Transport based on the command line arguments, listen on a TCP socket, and run as a daemon
+/// process serving network requests.  Success of the child is verified by means of a
+/// `SessionStartResult` JSON message sent through the standard output pipe.
+fn start_session(run_file_fn: impl FnOnce(u16) -> PathBuf) -> Result<Box<dyn Serialize>> {
+    let mut args = args_os();
+
+    let mut child = Command::new(args.nth(0).unwrap()) // Same executable
+        .arg("--child") // Add argument to let the new process know it is the daemon child
+        .args(args) // Propagate all existing arguments: --interface, etc.
```

Comment:
...and then this can be `env::args_os().skip(1)`.

Reply:
Done